File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    answer = 0
    dx = [0, 1 , 0, -1]
    dy = [1, 0, -1, 0]
    for i in range(4):
        h_check = h + dx[i]
        w_check = w + dy[i]
        if(h_check < 0 or h_check >= len(board) or w_check < 0 or w_check >= len(board[0])):
            continue
        elif(board[h_check][w_check] == board[h][w]):
            answer += 1
            
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.cpp
#include <string>
#include <vector>

using namespace std;
vector<int> dx = {0, 1, 0, -1};
vector<int> dy = {1, 0, -1, 0};

int solution(vector<vector<string>> board, int h, int w) {
    int answer = 0;
    for(int i = 0 ; i < 4 ; i++)
    {
        int h_check = h + dx[i];
        int w_check = w + dy[i];
        if(h_check < 0 || h_check >= board.size() || w_check < 0 || w_check >= board[0].size()){
            continue;
        }else if(board[h_check][w_check] == board[h][w]) {
            answer++;
        }
    }
    return answer;
}